File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        while(s.size() > 0 && s.find(part) < s.size()){
            int start = s.find(part);
            s.erase(start, part.size());
        }
        return s;
    }
};

File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.java
class Solution
{
    public String removeOccurrences(String s, String part)
    {
        while (s.indexOf(part) != -1)
        {
            int start = s.indexOf(part);
            s = s.substring(0, start) + s.substring(start + part.length());
        }

        return s;
    }
}